packages/eas-cli/src/build/build.ts

```diff
@@ -229,9 +234,16 @@ export function handleBuildRequestError(error: any, platform: Platform): never {
   throw error;
 }
 
-async function uploadProjectAsync<TPlatform extends Platform>(
-  ctx: BuildContext<TPlatform>
-): Promise<string> {
+async function uploadProjectAsync<TPlatform extends Platform>({
+  ctx,
+  generateMetadataFile = true,
```

Comment:
When do we not want to generate metadata file? I can see no `generateMetadataFile: false` anywhere. If that's so, let's remove this option altogether.

Also, a side suggestion — I think `should[Action]` reads better for boolean options than `[action]`. `generateMetadataFile` could be more than just a boolean. With `should…` it's clearer.